src/Queue/QueueFactory.php

```diff
@@ -57,20 +57,19 @@ public function getQueue(string $name): IQueue
 	 * @throws QueueFactoryException
 	 * @throws ConnectionFactoryException
 	 */
-	private function create(string $name): IQueue
+	private function create(string $name, bool $forceDeclare): IQueue
 	{
 		try {
 			$queueData = $this->queuesDataBag->getDataBykey($name);
 
 		} catch (\InvalidArgumentException $e) {
-
 			throw new QueueFactoryException("Queue [$name] does not exist");
 		}
 
 		// (ConnectionFactoryException)
 		$connection = $this->connectionFactory->getConnection($queueData['connection']);
 
-		if ($queueData['autoCreate']) {
+		if ($forceDeclare || $queueData['autoCreate']) {
 			$connection->getChannel()->queueDeclare(
```

Comment:
I would better like and approch of calling `$connection->getChannel()->queueDeclare` straight in the `DeclareQueuesAndExchangesCommand `